src/include/storage/lock.h

```diff
@@ -178,11 +179,16 @@ typedef struct LOCKTAG
  * rather than accessing the fields directly.  Note multiple eval of target!
  */
 
+/*
```

Comment:
we need to acquire locks in shared mode a two places, can we do tag manipulation there only in some way? It is not ideal to call get ENR for each call here. Also, let us see if there are some existing hooks we can use here (in case)